edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Core.Test/NestedEdgeParentUrlParserTest.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Xunit;
+
+    internal class MockEnvironment : IEnvironmentWrapper
+    {
+        public Dictionary<string, string> Map = new Dictionary<string, string>();
+
+        public Option<string> GetVariable(string variableName)
+        {
+            if (this.Map.ContainsKey(variableName))
+            {
+                return Option.Some(this.Map[variableName]);
+            }
+            else
+            {
+                return Option.None<string>();
+            }
+        }
+
+        public void SetVariable(string variableName, string value)
+        {
+            // Check for entry not existing and add to dictionary
+            this.Map[variableName] = value;
+        }
+    }
+
+    [Unit]
+    public class NestedEdgeParentUriParserTest
+    {
+        [Theory]
+        [InlineData("$upstream:9000/comp/ea:tag1", "parentaddress:9000/comp/ea:tag1", null, "IOTEDGE_GATEWAYHOSTNAME", "parentaddress")]
```

Comment:
>9000 [](start = 31, length = 4)

prob out of scope for this change, but I don't think we're tracking the change to enforce that port is omitted in parent_hostname, and that should probably be bundled with this feature area